media/js/mozorg/home/home.js

```diff
@@ -2,147 +2,203 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-(function(Waypoint) {
-    'use strict';
-
-    function getNextEl(el) {
-        el = el.nextSibling;
-        while (el) {
-            if (el.nodeType === 1) {
-                return el;
-            }
-            el = el.nextSibling;
-        }
-        return null;
-    }
+// YouTube API hook has to be in global scope
+ function onYouTubeIframeAPIReady() {
+     'use strict';
+
+     Mozilla.firstRunOnYouTubeIframeAPIReady();
+ }
+
+ (function(Waypoint) {
+     'use strict';
+
+     var tag = document.createElement('script');
+     tag.src = 'https://www.youtube.com/iframe_api';
+     var firstScriptTag = document.getElementsByTagName('script')[0];
+     firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
+    //
+     function onYouTubeIframeAPIReady() {
+
+        // lazy load video when visitor clicks the placeholder.
+         var videoLink = document.querySelector('.video-play');
+         var videoId = videoLink.getAttribute('data-id');
+
+         new YT.Player(videoLink, {
+             height: '703',
+             width: '1200',
+             videoId: videoId,
+             playerVars: {
+                 modestbranding: 1, // hide YouTube logo.
+                 rel: 0, // do not show related videos on end.
+             },
+             events: {
+                 'onReady': onPlayerReady,
+                 'onStateChange': onPlayerStateChange
+             }
+         });
+
+         function onPlayerReady(event) {
+             event.target.playVideo();
+         }
+
+         function onPlayerStateChange(event) {
+             var state;
+
+             switch(event.data) {
+             case YT.PlayerState.PLAYING:
+                 state = 'video play';
+                 break;
+             case YT.PlayerState.PAUSED:
+                 state = 'video paused';
+                 break;
+             case YT.PlayerState.ENDED:
+                 state = 'video complete';
+                 break;
+             }
```

Comment:
This is the function we want to use to track the video play/pause/end events. Right now ESLint is complaining, becuase the `state` variable is not being used here.